### DISCUSSION TITLE
Why isn't "allow-parallel-runners" used by default?

### DISCUSSION THREAD

**Discussion by leonklingele** [golangci/golangci-lint#4100] [Q&A]
I was wondering and couldn't find a reason why the golangci-lint config option `allow-parallel-runners` is not enabled by default. Is there any downside to not allow concurrent executions?

The locking mechanism to enforce sequential runs was introduced in https://github.com/golangci/golangci-lint/pull/812 while opting in to parallel runs was added in https://github.com/golangci/golangci-lint/pull/1070, both changes unfortunately without an explanation on why they were made.

**Comment by fastcat** [comment]
Was wondering the same thing and found this thread and clicked through the PRs and I think I found the answer:

Issue #108 is noted as being fixed by #812. That issue talks about golangci-lint invocations spawned from file-save operations in IDEs and file-watcher style contexts, where redundant copies could be very wasteful. E.g. if you save 10 open files all at once and that leads to 10 copies all linting the same project/packages.

**Reply by silverwind** [reply]
That's a bug in that editor integration, tbh.
